### PR TITLE
VideoPlayer: rewrite yuv to rgb conversion for OpenGL

### PR DESCRIPTION
**Pull request by FernetMenta**
- rewrite of YUV - RGB conversion for OpenGL
- move renderflags into renderer

### CODE REVIEW

**Comment by FernetMenta** [comment]
@popcornmix please adapt omxplayer

**Comment by Rechi** [comment]
wrong revision was build
jenkins build this please

**Comment by FernetMenta** [comment]
@afedchin you may want to rework Windows too. The old code is wrong, i.e. it does not consider transformations between colour primaries. Only HD videos had correct colous because bt.709 is identical with sRGB.

**Comment by FernetMenta** [comment]
jenkins build this please

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp [hunk 1]
```diff
@@ -199,7 +199,7 @@ CDVDVideoCodec::VCReturn CAddonVideoCodec::GetPicture(VideoPicture* pVideoPictur
     pVideoPicture->pts = static_cast<double>(picture.pts);
     pVideoPicture->dts = DVD_NOPTS_VALUE;
     pVideoPicture->color_range = 0;
-    pVideoPicture->color_matrix = 4;
+    pVideoPicture->color_space = 4;
```
Comment (FernetMenta) [inline xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp]: @peak3d why do you set color space to 4, which is: FCC Title 47 Code of Federal Regulations 73.682
does not make much sense to me

**Comment by afedchin** [comment]
@FernetMenta which is main change between old YUVMatrix and new implementation?

**Comment by FernetMenta** [comment]
@afedchin 
- conversion between colour primaries source / destination
- scaling for source > 8 bits

File: xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp (hunk 1 above)
Comment (peak3d) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp]: @FernetMenta cp/paste from some source, no idea what I do here, so pls. let me know how to evaluate this value.

Comment (peak3d) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp]: B.t.w. same code is in AML

Comment (FernetMenta) [reply xbmc/cores/VideoPlayer/DVDCodecs/Video/AddonVideoCodec.cpp]: @peak3d you should detect color space of the source and set value according to AVColorSpace (ffmpeg). If source has no color space set, set it to AVCOL_SPC_UNSPECIFIED

**Comment by MilhouseVH** [comment]
Could this be related? [forum](https://forum.kodi.tv/showthread.php?tid=298462&pid=2697410#pid2697410)

**Comment by FernetMenta** [comment]
possible

**Comment by koying** [comment]
Yep:
`21:01:00.334 T:1540745504   DEBUG: CRenderManager::UpdateResolution: mono`

**Pull request merged**